File: Leetcode/sbgo/Leetcode506-相对名次.go
package sbgo

import (
	"sort"
	"strconv"
)

type helper struct {
	score, idx int
}

var rec = [3]string{"Gold Medal", "Silver Medal", "Bronze Medal"}

func findRelativeRanks(score []int) []string {
	n := len(score)
	help := make([]helper, n)
	for i := 0; i < n; i++ {
		help[i].score = score[i]
		help[i].idx = i
	}
	sort.Slice(help, func(i, j int) bool {
		return help[i].score > help[j].score
	})
	res := make([]string, n)
	for i := 0; i < n; i++ {
		if i < 3 {
			res[help[i].idx] = rec[i]
		} else {
			res[help[i].idx] = strconv.Itoa(i + 1)
		}
	}
	return res
}

File: Leetcode/PY/Leetcode506-相对名次.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        help,n = [],len(score)
        for i in range(n):
            help.append([score[i],i])
        help.sort(key = lambda help : help[0])
        cnt = 1
        res = ['0'] * n
        for i in range(n - 1,-1,-1):
            if cnt == 1: res[help[i][1]] = "Gold Medal"
            elif cnt == 2: res[help[i][1]] = "Silver Medal"
            elif cnt == 3: res[help[i][1]] = "Bronze Medal"
            else: res[help[i][1]] = str(cnt)
            cnt += 1
        return res